Source/include/Ably/ARTPush.h

```diff
@@ -33,6 +33,13 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)didAblyPushRegistrationFail:(nullable ARTErrorInfo *)error;
 
+/**
+ Ably will call the implementation of this method when the activation process is completed with success, so you can request other types of device tokens, such as `location`, `pushtotalk` etc.
+ See https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns for possible values of `apns-push-type`.
+ Before making request to Apple for alternative device token you should check whether you've already did this after application launch.
+ */
+- (void)shouldRequestAlternativeDeviceToken;
```

Comment:
This method is the way to tell the user that the state machine is in the `ARTPushActivationStateWaitingForNewPushDeviceDetails` state (and can be used for tokens updating) which can't be known in any other way, since the state machine is not accessible to the user. `didActivateAblyPush:` is not a place for this as I've mentioned elsewhere.

So it shouldn't be attributed only to location pushes, since it should be used to request other tokens as well in the future. Imaging you need to request, say, `pushtotalk` token. Now you need two methods for that: `shouldRequestLocationDeviceToken` and `shouldRequestPTTDeviceToken`. But in what order you will call them for the user? You can't call one right after another without waiting the state machine to complete with at least one. Or you can, but in this case you need to explain user, that they should only save token on the server when both tokens will be fetched, and introduce `ARTPush.didRegister...` that takes an array of tokens as an argument (otherwise you will break your state machine sequence of states and events). But in this case why would you need two methods `shouldRequest...` in the first place? You can introduce just one where the user decides what particular token(s) they need. And we only provide a method to save the token (`ARTPush.didRegister...`) that we support. Once one token is saved, this method is called again, and the user can request another token, if any needed.

Also the goal of this solution was to introduce as few changes to the state machine code as possible to keep this PR mergable in a short perspective. And the selected approach with just one method for requesting tokens serves this purpose well IMHO.

Other things to keep in mind while reviewing this:

I consider different tokens as one single "combined" token that is stored in the "push" field. This way state machine doesn't know anything about different types of tokens (other than user facing API) and can be kept almost unchanged. Also I decided not to change the format tokens are stored locally other than the storage key names.

Reply:
> So it shouldn't be attributed only to location pushes, since it should be used to request other tokens as well in the future.

That's fine — my point is just that the SDK _currently_ is only able to handle location tokens, so for the documentation for this method to say "so you can request other types of device tokens, such as `location`, `pushtotalk` etc" is a bit confusing, since it suggests that the SDK is actually able to handle a `pushtotalk` token for example. I'm just saying that we should be clear that the only currently supported token type is `location`.

Reply:
> Also we should tell the user what they're meant to do with the token once they've got it.

And here I mean that the documentation for `shouldRequestAlternativeDeviceToken` should tell the user that once they've fetched the location token in response to receiving this method call, they should pass it to `didRegisterForLocationNotificationsWithDeviceToken:`.

Reply:
After further thinking about this new callback, I, at first, decided to [remove](https://github.com/ably/ably-cocoa/pull/1771/commits/b2bffc48921cc77956cc1d314839ee0aeba9d4cd) its repeating calls after each update and make it once per `activate()` call (and later, if needed, introduce `ARTPush` class method to save an array of device tokens simultaneously). And then I've checked why `didActivateAblyPush:` wasn't a good fit for this in the first place. And turned out that it actually is. I wasn't able to find any issues while using it. Last week, for some reason, I've made an erroneous conclusion that it's not a place to request or save another token. So I've removed this new callback entirely. Please, see [commit](https://github.com/ably/ably-cocoa/pull/1771/commits/cdc6b94b0c18d67ff753af1655483f886f7a9af0) message for more details.

I'll refer this comment in other messages re `shouldRequestAlternativeDeviceToken` method.